### ISSUE TITLE
API to verify header before importing

### ISSUE THREAD

**Issue by svyatonik** [paritytech/parity-bridges-common#421]
Case:
1) relay is currently configured to have at most 8 submit-header-transactions in the-wild;
2) imagine that these transactions are H10...H17;
3) imagine that H11 requires justification (after #409) because it enacts new authorities set;
4) if relay would submit all 8 transactions, transactions H12...H17 would be rejected by the pallet => relay would lose its funds.

The solution that has been used in sub-2-eth sync is to provide API that given `Vec<Header>` would return headers that the contract will be able to import (given its current state). So relay just haven't sent H12...H17 until justification for H11 was also provided and mined.

cc @HCastano

**Comment by svyatonik** [comment]
superseded by #761
